### PR TITLE
Perception: support multi-class detection for PointPillars

### PR DESCRIPTION
**Pull request by jeroldchen**
Perception: simplified computation of bounding box's params
Perception: support multi-class detection for PointPillars
Perception: update PointPillars models and remove useless ones

### CODE REVIEW

File: modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.h [hunk 1]
```diff
@@ -118,6 +118,7 @@ class PointPillars {
   const int NUM_ANCHOR_Y_INDS_;
   const int NUM_ANCHOR_R_INDS_;
   const int NUM_ANCHOR_;
+  const int NUM_CLASS_;
   const int RPN_BOX_OUTPUT_SIZE_;
   const int RPN_CLS_OUTPUT_SIZE_;
   const int RPN_DIR_OUTPUT_SIZE_;
```
Comment (jinghaomiao) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.h:124]: static const variable: use kConstVariable naming convension.

File: modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_test.cc [hunk 2]
```diff
@@ -469,10 +473,13 @@ TEST(TestSuite, CheckDoInference) {
   test_obj.pclToArray(pcl_pc_ptr, points_array, NORMALIZING_FACTOR);
 
   std::vector<float>* out_detections(new std::vector<float>());
-  test_obj.DoInference(points_array, pcl_pc_ptr->size(), out_detections);
+  std::vector<int>* out_labels(new std::vector<int>());
```
Comment (jinghaomiao) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_test.cc:476]: Use smart pointers such as uniq_ptr or shared_ptr for safe memory management. Please fix it for all pointer usages.

Comment (jeroldchen) [reply modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_test.cc:476]: OK. I will refactor part of the code cause it cannot meet our requirements for customizing some of the params. The above problems will be resolved during the refactoring.

**Comment by jinghaomiao** [comment]
@gchen-apollo @weidezhang Please do the code review.

**Comment by gchen-apollo** [comment]
> @gchen-apollo @weidezhang Please do the code review.

sure, wait for update and new pull from @jeroldchen

**Comment by jeroldchen** [comment]
@jinghaomiao @gchen-apollo I have applied the changes you required.

File: modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc [hunk 3]
```diff
@@ -42,82 +42,87 @@ namespace perception {
...
+      kMinXRange(0.0f),
+      kMinYRange(-39.68f),
+      kMinZRange(-3.0f),
+      kMaxXRange(69.12f),
+      kMaxYRange(39.68f),
+      kMaxZRange(1),
+      kBatchSize(1),
+      kNumIndsForScan(512),
+      kNumThreads(64)
+      // if you change kNumThreads, need to modify NUM_THREADS_MACRO in
```
[the remark this answers is not in the document]
Comment (jeroldchen) [reply modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc:84]: 但这些变量是常量，之所以放在构造函数里初始化，而不是在头文件里初始化，是因为有些变量依赖其他变量的值来初始化，后面这些变量的值一直都不会改变了；而且我要是改成variable_的形式，lint check就会fail，因为有int array[variable_]; 这样的初始化，lint会提示我改成int array[kVariable];

[the remark this answers is not in the document]
Comment (gchen-apollo) [reply modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc:84]: 这个lint问题第一次遇到。如果这样，把一部分转成static const吧。

**Comment by jeroldchen** [comment]
To specify a semantic constraint, I declare hyperparameter variables static const. They are named with 'k' followed by CamelCase. A new commit has just pushed to apply these changes.

**Pull request merged**